### DISCUSSION TITLE
Budget limit / Spending limit

### DISCUSSION THREAD

**Discussion by maiieul** [planetscale/discussion#417] [Feature Requests] [answered]
Hi guys,

Is Planetscale planning on implementing a budget limit for the scaler and team plans?

As stated [many times on Hacker News]( https://news.ycombinator.com/item?id=30457960) when you increased the free tier to 1 billion row reads, a pricing model based on row reads and row writes without a budget limit can feel a bit scary. 

Haven't you had customers that wrote a query on a large dataset reading millions of rows, and executing it many times?

Haven't you had customers that had to pay additional fees because they ran into spam issues?

I think that these are some legitimate questions and concerns customers and potential customers can have, and therefore a budget limit could increase the adoption of Planetscale. CockroachDB has highlighted this feature on its [pricing page](https://www.cockroachlabs.com/pricing/). Wouldn't this be possible with the way Planetscale works?

**Comment by fooforge** [comment, marked as answer]
Hey @maieulChevalier,

The short answer is yes. There is always room for improvement in enabling users to better control their spent, and I agree that it would likely ease people's minds a bit. We've got some ideas in this regard that we're hoping to get to over the course of this year.

The longer answer entails that we don't see many people struggling with this much. The limits were designed with ample room for experimentation so that you don't have to worry about each query you run against your database. Most users do not reach or exceed these limits.

If you have racked up more reads and/or writes than what's included in your plan due to a mistake (e.g. insufficient indexes) or lack of knowledge, [open a ticket](https://support.planetscale.com/) with us and we'll figure it out, and help optimize your schema to achieve lower reads and writes (and often better performance). The two scenarios that you have described fall into that area.
In each billing cycle, we also keep a close eye on accounts that have seen a large increase in their spent. If so, we reach out proactively to learn about their situation. I cannot disclose numbers, but what I can share is that it's taking up a very minimal amount of my time and that I don't remember a single case where a user has expressed dissatisfaction.

I have said this before on this discussion board, but I'll happily repeat myself:
Software development is a vast field, it's complex and developers make mistakes. We have no desire in exploiting any of this complexity or the people who are just getting started with databases, but rather aim to grow along with you and your business. That requires trust and we would need less of it with better cost control tooling, for sure. But since we, as a company, cannot build out everything all at once, we rely on your trust in us while we're building out this platform.

I hope that the above insights have helped reinforcing that trust at least a little bit. ;-)

**Reply by maiieul** [reply]
Wow, thanks for the quick and extensive response, and thanks for your clarity and transparency. This is reassuring indeed.
